modules/genai-ecosystem/pages/agent-neo.adoc

```diff
@@ -1,90 +1,116 @@
+* Injecting session UUIDs generated on the frontend, integrating these with LangSmith LLM logging, and writing back LangSmith metadata statistics into the graph (Pending)^[5]^.
+* Capturing user response feedback through the UI and incorporating this into the graph to potentially inform response ranking and analyze application satisfaction (Pending)^[6]^.
 
-TODO
-The front-end is a React/Streamlit Application?? and the Backend 
-The frontend is a React Application and the backend a Python FastAPI.
-It uses the https://python.langchain.com/docs/use_cases/graph/constructing[llm-graph-transformer module^] that we contributed to LangChain.
+== Why Graph RAG?
 
+Agent Neo utilizes Graph-RAG technology to offer significant advantages:
 
-== Functionality Includes
+* *Completeness of Responses*: Combines classical vector similarity searches with graph traversals to capture a broader context around text similarities beyond simple metric comparisons.
+* *Explainability & Observability*: Enhances the transparency and explainability of operations by incorporating vectors within a graph structure, moving away from the "black box" nature of traditional models.
+* *Rapid Iteration and Integration*: Facilitates quick iterations and seamless integration of diverse data sources through Neo4j’s Label Property Graph, supported by in-platform feature engineering capabilities.
 
-* import source documents into the graph database
-* chunk text documents
-* generate text enbeddings and vector index them
-// * extract entities?
-* connect similar chunks via scored similarity relationships
-* employ GraphRAG to retrieve relevant chunks, their documents and related content for answering questions
-* store conversations (questions and answers) in the graph and connect them back to the originating context
+image::agent-neo.png[Agent Neo Architecture, width=600, align="center"]
 
-image::agent-neo-structure.png[width=600, align=center]
+== Installation
 
-== How it works
+=== Prerequisites
 
-1. Uploaded Sources are stored as `Document` nodes in the graph
-2. Each document (type) is loaded with the LangChain Loaders
-3. The content is split into Chunks 
-4. Chunks are stored in the graph and connected to the Document and to each other for advanced RAG patterns
-5. Highly similar Chunks are connected with a `SIMILAR` relationship to form a kNN Graph
-6. Embeddings are computed and stored in the Chunks and Vector index
-7. Using the llm-graph-transformer or diffbot-graph-transformer entities and relationships are extracted from the text
-8. Entities are stored in the graph and connected to the originating Chunks
+* Docker installed on your machine
+* Python 3.8 or higher
+* Node.js 12 or higher
 
+=== Setup Guide
 
-// TODO architecture diagram
+To set up Agent Neo, execute the following steps:
 
-== Relevant Links
-[cols="1,4"]
-|===
-| Online | https://agent-neo.neo4jlabs.com/
```

Comment:
can we keep the structure so it's consistent with what we have on the other pages?